packages/react-native/React/Views/RCTCursor.m

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// [macOS]
+
+#import <Foundation/Foundation.h>
+#import "RCTCursor.h"
+
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 150000 /* __MAC_15_0 */
+#define RCT_MAC_OS_15_SDK_AVAILABLE
+#endif // __MAC_OS_X_VERSION_MAX_ALLOWED
+
+#if TARGET_OS_OSX
+NSCursor *NSCursorFromRCTCursor(RCTCursor cursor)
+{
+  NSCursor *resolvedCursor = nil;
+  switch (cursor) {
+    case RCTCursorAuto:
+      break;
+    case RCTCursorAlias:
+      resolvedCursor = [NSCursor dragLinkCursor];
+      break;
+    case RCTCursorAllScroll:
+      // Not supported
+      break;
+    case RCTCursorCell:
+      // Not supported
+      break;
+    case RCTCursorColResize:
+#ifdef RCT_MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor columnResizeCursor];
+      } else {
+        resolvedCursor = [NSCursor resizeLeftRightCursor];
+      }
+#else
+      resolvedCursor = [NSCursor resizeLeftRightCursor];
+#endif // RCT_MAC_OS_15_SDK_AVAILABLE
+      break;
+    case RCTCursorContextMenu:
+      resolvedCursor = [NSCursor contextualMenuCursor];
+      break;
+    case RCTCursorCopy:
+      resolvedCursor = [NSCursor dragCopyCursor];
+      break;
+    case RCTCursorCrosshair:
+      resolvedCursor = [NSCursor crosshairCursor];
+      break;
+    case RCTCursorDefault:
+      resolvedCursor = [NSCursor arrowCursor];
+      break;
+    case RCTCursorEResize:
+#ifdef RCT_MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor frameResizeCursorFromPosition:NSCursorFrameResizePositionRight
+                                                    inDirections:NSCursorFrameResizeDirectionsOutward];
+      } else {
+        resolvedCursor = [NSCursor resizeRightCursor];
+      }
+#else
+      resolvedCursor = [NSCursor resizeRightCursor];
+#endif // RCT_MAC_OS_15_SDK_AVAILABLE
+      break;
+    case RCTCursorEWResize:
+#ifdef RCT_MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor frameResizeCursorFromPosition:NSCursorFrameResizePositionLeft
+                                                    inDirections:NSCursorFrameResizeDirectionsAll];
+      }
+#endif // RCT_MAC_OS_15_SDK_AVAILABLE
+      break;
+    case RCTCursorGrab:
+      resolvedCursor = [NSCursor openHandCursor];
+      break;
+    case RCTCursorGrabbing:
+      resolvedCursor = [NSCursor closedHandCursor];
+      break;
+    case RCTCursorHelp:
+      // Not supported
+      break;
+    case RCTCursorMove:
+      // Not supported
+      break;
+    case RCTCursorNEResize:
+#ifdef RCT_MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
```

Comment:
not sure if the `@` is similar to C# to escape keywords or if this is a compiler instruction in swift since it looks like a runtime check., but we have the same expression duplicated a lot... I realize it is inside a case statement, but shouldn't this check be a 'constant' and therfore lifted maybe to a member / static?
Ignore me if this is compiler interpreted

Reply:
It's a runtime check in objective c: https://developer.apple.com/documentation/swift/marking-api-availability-in-objective-c

Reply:
Spoke with Tommy offline. This is a common pattern in apple to use these, making it easily searchable whenever you want to update minimum OS deployments. The real ugliness comes from the ifdefs, which I can't fix easily :P.

Reply:
The compiler also checks the code against available SDKs and throws errors. You cannot make it a constant, but we can probably refactor this function into two where we only do the check once in the first, and fallback to the second. I'm not going to block this PR though since this is a really common pattern.